### PR TITLE
Consider the original file path for renamed files when checking file regexp matching for Gerrit.

### PR DESCRIPTION
**Pull request by cjwagner**
/assign @chaodaiG

### CODE REVIEW

File: prow/gerrit/client/client.go [hunk 1]
```diff
@@ -684,3 +685,21 @@ func (h *gerritInstanceHandler) QueryChangesForProject(log logrus.FieldLogger, p
 		}
 	}
 }
+
+// ChangedFilesProvider lists (in lexicographic order) the files changed as part of a Gerrit patchset.
+// It includes the original paths of renamed files.
+func ChangedFilesProvider(changeInfo *ChangeInfo) config.ChangedFilesProvider {
```
Comment (cjwagner) [inline prow/gerrit/client/client.go:691]: Let me know if you'd prefer this helper function lived in the source.go file instead. Wasn't sure where it made the most sense.

Comment (chaodaiG) [reply prow/gerrit/client/client.go:691]: `source` package is so far very lightweight functions that's safe to be imported by any package, such as `config`. I'm not seeing this function being that widely used, so let it be here should be fine

File: prow/gerrit/client/client.go [hunk 2]
```diff
@@ -684,3 +685,21 @@ func (h *gerritInstanceHandler) QueryChangesForProject(log logrus.FieldLogger, p
...
+	return func() ([]string, error) {
+		changed := sets.NewString()
+		revision := changeInfo.Revisions[changeInfo.CurrentRevision]
```
Comment (chaodaiG) [inline prow/gerrit/client/client.go:694]: this function takes in pointer, should we check nil pointer here?

**Review by chaodaiG — APPROVED** [review]
/hold

File: prow/gerrit/client/client.go (hunk 2 above)
Comment (cjwagner) [reply prow/gerrit/client/client.go:694]: Will fix in the next PR I'm making. 
/hold cancel

**Pull request merged**